### PR TITLE
[cli] Prioritize `index.html` over other files named index

### PR DESCRIPTION
**Pull request by MaxLeiter**
Mirrors production more closely. Initially reported [on spectrum](https://spectrum.chat/zeit/now/statically-served-default-document-is-the-first-index-file-not-index-html~0ad66f8a-a139-4a0e-94d9-7e5ceb0fa043)

### CODE REVIEW

**Review by TooTallNate — COMMENTED** [review]
Code seems good! This will need a regression test (you can add one to `test/dev-server.unit.js`).

File: packages/now-cli/test/dev-server.unit.js [hunk 1]
```diff
@@ -498,3 +498,12 @@ test('[DevServer] parseListen()', t => {
   t.truthy(err);
   t.is(err.message, 'Unknown `--listen` scheme (protocol): bad:');
 });
+
+test(
+  '[DevServer] index.html prioritized over other index.* files',
+  testFixture('now-dev-prioritize-index-html', async (t, server) => {
+    const res = await fetch(`${server.address}`);
+    const html = await res.text();
+    t.is(html.includes(`<title>Hello World</title>`) > -1, true);
```
Comment (AndyBitz) [inline packages/now-cli/test/dev-server.unit.js]:
```suggestion
    t.truthy(html.includes('<title>Hello World</title>'));
```

`.includes` already returns a boolean

File: packages/now-cli/src/util/dev/server.ts [hunk 2]
```diff
@@ -2014,11 +2013,26 @@ async function findBuildMatch(
...
+        }
+      }
     }
   }
+
+  // check for a non-.html index file
+  if (typeof bestIndexMatch !== 'undefined') {
+    return bestIndexMatch;
+  }
   return null;
```
Comment (AndyBitz) [inline packages/now-cli/src/util/dev/server.ts]:
```suggestion
  return bestIndexMatch || null;
```

File: packages/now-cli/test/fixtures/unit/now-dev-prioritize-index-html/now.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "version": 2,
+  "name": "now-dev-prioritize-index-html"
+}
```
Comment (styfle) [inline packages/now-cli/test/fixtures/unit/now-dev-prioritize-index-html/now.json]: You can remove this file, its not necessary

File: packages/now-cli/test/dev-server.unit.js [hunk 4]
```diff
@@ -498,3 +498,12 @@ test('[DevServer] parseListen()', t => {
   t.truthy(err);
   t.is(err.message, 'Unknown `--listen` scheme (protocol): bad:');
 });
+
+test(
+  '[DevServer] index.html prioritized over other index.* files',
+  testFixture('now-dev-prioritize-index-html', async (t, server) => {
```
Comment (styfle) [inline packages/now-cli/test/dev-server.unit.js]: Let's move this test to `test/dev/integration.js` so we can compare with production.

We also need to check that `/index.css` returns the correct content too.

Take a look at this this test for example: https://github.com/vercel/vercel/blob/8ec8affe29e49c27c437d6d74daa8a4f343fce94/packages/now-cli/test/dev/integration.js#L472-L482

**Review by TooTallNate — DISMISSED** [review]

**Review by TooTallNate — APPROVED** [review]

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**